src/controllers/restatedeployment/controller.rs

```diff
@@ -437,75 +458,161 @@ impl RestateDeployment {
+                    };
 
-                status_from_replica_set(
-                    self.spec.replicas,
-                    &mut rsd_status,
-                    current_replicaset.status.as_ref(),
-                );
+                    status_from_replica_set(
+                        self.spec.replicas,
+                        &mut rsd_status,
+                        current_replicaset.status.as_ref(),
+                    );
 
-                (
-                    Ok(action),
-                    "RestateDeployment is deployed".into(),
-                    "Deployed".into(),
-                    "True".into(),
-                )
-            }
-            Err(Error::DeploymentNotReady {
-                message,
-                reason,
-                requeue_after,
-                replica_set_status,
-            }) => {
-                let requeue_after = requeue_after.unwrap_or(Duration::from_secs(60));
-
-                status_from_replica_set(
-                    self.spec.replicas,
-                    &mut rsd_status,
-                    replica_set_status.as_deref(),
-                );
+                    if let Some(id) = current_replicaset
+                        .annotations()
+                        .get(RESTATE_DEPLOYMENT_ID_ANNOTATION)
+                    {
+                        rsd_status.deployment_id = Some(id.clone());
+                    }
 
-                (
-                    Ok(Action::requeue(requeue_after)),
+                    (
+                        Ok(action),
+                        "RestateDeployment is deployed".into(),
+                        "Deployed".into(),
+                        "True".into(),
+                    )
+                }
+                Err(Error::DeploymentNotReady {
                     message,
                     reason,
-                    "False".into(),
-                )
-            }
-            Err(Error::HashCollision) => {
-                rsd_status.collision_count = Some(rsd_status.collision_count.unwrap_or(0) + 1);
+                    requeue_after,
+                    replica_set_status,
+                }) => {
+                    let requeue_after = requeue_after.unwrap_or(Duration::from_secs(60));
+
+                    status_from_replica_set(
+                        self.spec.replicas,
+                        &mut rsd_status,
+                        replica_set_status.as_deref(),
+                    );
 
-                (
+                    (
+                        Ok(Action::requeue(requeue_after)),
+                        message,
+                        reason,
+                        "False".into(),
+                    )
+                }
+                Err(Error::HashCollision) => {
+                    rsd_status.collision_count = Some(rsd_status.collision_count.unwrap_or(0) + 1);
+
+                    (
                     // requeue immediately, although in practice the status update should requeue us
```

Comment:
i think this comment may no longer be true if we're gonna stop watching our own status